### PR TITLE
add followers. add following. add starred repos.

### PR DESCRIPTION
**Pull request by blackknight75**
@s-espinosa
This is a general pull request. I am following the development pattern that we worked on and wanted to make sure I'm on the right track. Made some big breakthroughs in understanding today.

### CODE REVIEW

File: app/controllers/following_controller.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class FollowingController < ApplicationController
+
+  def index
+    @followings = Following.all(current_user)
+    # binding.pry
```
Comment (s-espinosa) [inline app/controllers/following_controller.rb:5]: Be sure to get these out when you commit/push.

File: app/models/following.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    @image = params[:avatar_url]
+    @url = params[:html_url]
+  end
+
+  def self.all(current_user)
+    following_hash = GithubService.following(current_user)
+    following_hash.map do |following|
+      Following.new(following)
+    end
+  end
```
Comment (s-espinosa) [inline app/models/following.rb:15]: Like this pattern.

File: app/models/starred.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  def initialize(params)
+    @name = params[:name]
+    @description = params[:description]
+    @language = params[:language]
+    @updated_at = params[:updated_at]
+  end
+
+  def self.all(current_user)
+    starred_repos_hashes = GithubService.starred(current_user)
+    starred_repos_hashes.map do |starred|
```
Comment (s-espinosa) [inline app/models/starred.rb:13]: Still wish we had a better name than `starred_repos_hashes`. I think just drop `hashes` and leave it as `starred_repos`.

File: app/services/github_service.rb [hunk 4]
```diff
@@ -9,4 +9,14 @@ def self.followers(current_user)
     response = Faraday.get("https://api.github.com/users/#{current_user.username}/followers?client_id=#{ENV['github_client_id']}&client_secret=#{ENV['github_client_secret']}")
     JSON.parse(response.body, symbolize_names: true)
   end
+
+  def self.following(current_user)
+    response = Faraday.get("https://api.github.com/users/#{current_user.username}/following?cliend_id=#{ENV['client_id']}&client_secret=#{ENV['client_secret']}")
```
Comment (s-espinosa) [inline app/services/github_service.rb:14]: Take a look at the Faraday docs. You don't need to roll this URL by hand. You can set this as almost an attribute of the connection.

**Review by s-espinosa — COMMENTED** [review]
@blackknight75 Started this earlier today, and hadn't hit the button. Generally liking the code in this commit!

File: app/models/starred.rb (hunk 3 above)
Comment (blackknight75) [reply app/models/starred.rb:13]: Agreed. I'm gonna do a refactor later tonight that should get rid of these horrible names.

File: app/services/github_service.rb (hunk 4 above)
Comment (blackknight75) [reply app/services/github_service.rb:14]: @s-espinosa I saw some examples where people were setting instance variables for the first part of the url and then tacking on the endpoint in the class method. I'm going to add an example.

initialize
@connection = Faraday.new('https://api.github.com')

def self.repos(current_user)
  parse(connection.get("/users/#{current_user.username}/repos?client_id=#{ENV['github_client_id']}&client_secret=#{ENV['github_client_secret']}")
end

**Comment by blackknight75** [comment]
@s-espinosa I really appreciate the feedback Sal!

**Pull request merged**